pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -657,8 +657,11 @@ func (os *OpenStack) volumeService(forceVersion string) (volumeService, error) {
 		if autodetectedVersion := doBsApiVersionAutodetect(availableApiVersions); autodetectedVersion != "" {
 			return os.volumeService(autodetectedVersion)
 		} else {
-			// Nothing suitable found, failed autodetection
-			return nil, errors.New("BS API version autodetection failed.")
+			// Nothing suitable found, failed autodetection, just exit with appropriate message
+			err_txt := "BlockStorage API version autodetection failed. " +
+				"Please set it explicitly in cloud.conf in section [BlockStorage] with key `bs-version`"
+			glog.Fatalf(err_txt)
```

Comment:
Right, that's to make the compiler happy since we don't have a return statement at the end of the method